src/Nethermind/Nethermind.Core/ChainLevelInfo.cs

```diff
@@ -72,6 +72,19 @@ public BlockInfo? BeaconMainChainBlock
             return null;
         }
 
+        public int FindBeaconMainChainIndex()
+        {
+            for (int i = 0; i < BlockInfos.Length; i++)
+            {
+                if (BlockInfos[i].IsBeaconMainChain)
+                {
+                    return i;
+                }
+            }
+
+            return -1;
```

Comment:
use a const instead of a magic number?

```csharp
private const int NotFound = -1;
```

Reply:
Changed to notfound and then replaced with nullable. I like approach when we return -1 as index for not found item. But it does not force dev to check it.

Reply:
Returned notfound in other form of usage :D